hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java

```diff
@@ -786,6 +786,7 @@ private synchronized int readBuffer(ReaderStrategy reader, int len,
     while (true) {
       // retry as many times as seekToNewSource allows.
       try {
+        DFSClientFaultInjector.get().fetchFromDatanodeException();
```

Comment:
what does this do?

Reply:
this is used so that in unit test we can inject an error. we will mock fetchFromDatanodeException() in unit test, to throw IOException.